ndk/src/events/kinds/nip66/relay-monitor.ts

```diff
@@ -0,0 +1,397 @@
+        const kinds: boolean = this?.kinds?.length > 0 ? true : false;
+        const valid: boolean = frequency && checks && kinds;
+        return valid;
+    }
+
+    /**
+     * Checks if the current `RelayMonitor` instance is active based on whether it has published a single
+     * event within the criteria defined as "interval" in the Relay Monitor's registration event (kind: 10166) 
+     * 
+     * @returns {Promise<boolean>} Promise resolves to a boolean indicating whether the `RelayMonitor` is active.
+     * 
+     * @public
+     * @async
+    */
+    async isMonitorActive(): Promise<boolean> {
+        if(typeof this.active !== 'undefined') return this.active;
+        
+        const kinds: NDKKind[] = [];
+        if(this.kinds.includes(NDKKind.RelayDiscovery)) { 
+            kinds.push(NDKKind.RelayDiscovery);
+        }
+        if(this.kinds.includes(NDKKind.RelayMeta)) { 
+            kinds.push(NDKKind.RelayMeta);
+        }
+
+        const filter: NDKFilter = this.nip66Filter(kinds, { limit: 1 } as NDKFilter);
+
+        return new Promise((resolve, reject) => {
+            this.ndk?.fetchEvents(filter)
+                .then(events => {
+                    this.active = Array.from(events).length > 0;
+                    resolve(this.active);
+                })
+                .catch(reject);
+        });
+    }
+
+    /**
+     * Determines if the current monitor instance meets specified criteria.
+     * 
+     * @param criterias - The criteria to be matched against the monitor's capabilities.
+     * @returns {boolean} - True if the monitor meets all the specified criteria; otherwise, false.
+     * @public
+     */
+    public meetsCriterias( criterias: RelayMonitorCriterias ): boolean {
+        return RelayMonitor.meetsCriterias(this, criterias);
+    }
+
+    /**
+     * @description Reduces a set of `NDKEvent` objects to a list of relay strings.
+     * 
+     * @param {Set<RelayDiscovery | RelayMeta | NDKEvent>} events A set of `NDKEvent` objects.
+     * @returns Promise resolves to a list of relay strings or undefined.
+     * 
+     * @protected
+     */
+    protected reduceRelayEventsToRelayStrings( events: Set<RelayDiscovery | RelayMeta | NDKEvent> ): RelayListSet {
+        if(typeof events === 'undefined') {
+                return new Set() as RelayListSet;
+        }
+        return new Set(Array.from(events)
+            .map( event => {
+                return event.tags
+                    .filter( tag => tag[0] === 'd')
+                    .map( tag => tag[1] )[0];
+            })
+        );
+    }
+
+    /**
+     * Produces warnings (console.warn) if the monitor is invalid or has not been checked for validity.
+     * 
+     * @returns void
+     * 
+     * @private
+     */
+    protected maybeWarnInvalid(): void {
+        if( !this.isMonitorValid() ) {
+            console.warn(`[${this.pubkey}] RelayMonitor has not published a valid or complete "RelayMonitor" event.`);
```

Comment:
I try to stay away from writing to the console directly; at least here I would allow the caller to set a destination of the output (like passing in a `debug` interface)